src/DataDepsPaths.jl

```diff
@@ -4,22 +4,30 @@ using DataDeps: register, AbstractDataDep
 using FilePathsBase
 
 
+struct Satisfier
+    satisfies # DataDepPath → Bool
+    resolve # () → AbstractPath
+end
+Satisfier((satisfies, resolve)) =
+    Satisfier(satisfies, resolve)
+
 struct DataDependency <: AbstractDataDep
     name::String
     message::String
     satisfiers::Vector{Satisfier}
 end
 
-struct Satisfier
-    satisfies # DataDepPath → Bool
-    resolve # () → AbstractPath
-end
-
 struct DataDepPath # Should we actually subtype AbstractPath here?
     name::String
     parts::Vector{String}
 end
 
+struct DataDepPathRegistration
```

Comment:
Please forgive my unfamiliarity with DataDeps (today is pretty much my first usage of it): should this just be merged into `DataDependency` so that we can get the `register` dispatch from `DataDeps`?

Reply:
My current thoughts are to keep DataDepsPaths's API mostly sepertate from the current DataDeps API til is it working.
then make a plan for what to do with the current more basic DataDeps.